### PR TITLE
fix output corruption with more than 8K bytes and no newline

### PR DESCRIPTION
**Pull request by grondo**
Fix the issue described in #114.

Turns out pdsh doesn't handle output without newlines too well, and output flush was reading from uninitialized memory for anything >= 8K bytes.

This PR also fixes arbitrary addition of newlines to output during the final flush.

### CODE REVIEW

File: src/pdsh/dsh.c [hunk 1]
```diff
@@ -573,18 +564,39 @@ static int _do_output (int fd, cbuf_t cb, out_f outf, bool read_rc, thd_t *t)
...
-            outf ("%S: %s\n", t->host, buf);
-        else
-            outf ("%s\n", buf);
+    _flush_lines (cb, outf, false, t);
+
+    /* In case no newline at end of buffer, grab the rest of data */
+    while ((n = cbuf_read (cb, buf, sizeof (buf) - 1)) > 0) {
+        buf[n] = '\0';
+        if (t->labels && !labeled)
+            outf ("%S: ", t->host);
```
Comment (watson6282) [inline src/pdsh/dsh.c]: Shouldn't there be a "labeled = true;" inside this "if" statement? Otherwise, labeled is always false and so pointless.

Comment (grondo) [reply src/pdsh/dsh.c]: yes, that is correct. Good catch, and thanks for reviewing.

Comment (grondo) [reply src/pdsh/dsh.c]: Ok, added a fix and new test. If it looks ok after another review I can squash the fixup commit.

Comment (watson6282) [reply src/pdsh/dsh.c]: This looks good to me now.

**Comment by smemsh** [comment]
looking forward to merge :-)

**Comment by grondo** [comment]
Oops, I realized I still had some fixup! commits in this PR. Autosquashed them and rebased against current master.

**Comment by grondo** [comment]
BTW, @garlick or @morrone, could either one of  you enable Travis builds for this repo? I think you are the only 2 admins of the organization that would have access to do so.

**Comment by garlick** [comment]
Just gave you admin access to the pdsh repo.

**Comment by grondo** [comment]
Thanks @garlick! Pushed again to force a travis-ci build.

**Comment by grondo** [comment]
Thanks!

**Pull request merged**